### PR TITLE
Pass kwargs in apply_gradients to wrapped optimizer

### PR DESCRIPTION
**Pull request by andreABbauer**
This allows usage of tfa.optimizers.DecoupledWeightDecayExtension optimizers like AdamW with the apply_gradients parameter "decay_var_list" to select decay vars. 
This is a common application for AdamW since weight decay on BatchNorm params or biases is not always intended.

### CODE REVIEW

**Comment by reedwm** [comment]
I'm a little worried about passing `kwargs` through `merge_call` and `call_for_each_replica`. Distribute Strategies often will modify inputs in strange ways when passed to one of these functions, which is why we use a `_UnwrapPreventer` for the variables when calling `call_for_each_replica`. But there is no trick equivalent to `_UnwrapPreventer` for `merge_call`.

The distribute strategy team is working on deprecating `merge_call` and removing it from Optimizer and LossScaleOptimizer, so I think it's best to just wait until they do so before merging this PR. They plan on doing this in a few weeks. Alternatively, if you want this in sooner, please manually test this change with a DecoupledWeightDecayExtension optimizer where you pass `decay_var_list` with a MirroredStrategy to confirm that it works correctly. Unfortunately you cannot add a TF test itself which does this since it uses TF-addons.

**Comment by gbaned** [comment]
@andreABbauer Can you please check @reedwm's comments and resolve conflicts?. Thanks!

**Comment by gbaned** [comment]
@andreABbauer Any update on this PR? Please. Thanks!
